tests/unit/models/validators/test_validate.py

```diff
@@ -0,0 +1,147 @@
+import os
+from enum import Enum
+from typing import Any
+
+import pytest
+
+from scanomatic.generics.model import Model
+from scanomatic.io.jsonizer import dumps
+from scanomatic.models.analysis_model import (
+    AnalysisModelFields,
+    GridModelFields
+)
+from scanomatic.models.factories.analysis_factories import (
+    AnalysisModelFactory,
+    GridModelFactory
+)
+from scanomatic.models.factories.fixture_factories import (
+    FixtureFactory,
+    FixturePlateFactory,
+    GrayScaleAreaModelFactory
+)
+from scanomatic.models.fixture_models import FixtureModelFields
+from scanomatic.models.validators.validate import (
+    get_invalid,
+    get_invalid_as_text,
+    get_invalid_names,
+    validate
+)
+
+
+@pytest.mark.parametrize('model,updates,expect', (
+    (GridModelFactory.create(), {}, True),
+    (GridModelFactory.create(), {'use_utso': 5}, False),
+    (GridModelFactory.create(gridding_offsets=[[2, 3, 4]]), {}, False),
+    (AnalysisModelFactory.create(), {}, False),
+    (
+        AnalysisModelFactory.create(
+            compilation=os.path.abspath(__file__),
+            grid_model=GridModelFactory.create(),
+        ),
+        {},
+        True,
+    ),
+    (
+        AnalysisModelFactory.create(
+            compilation=os.path.abspath(__file__),
+            grid_model=GridModelFactory.create(gridding_offsets=((2, 3, 4),)),
+        ),
+        {},
+        False,
+    ),
+    (
+        FixtureFactory.create(
+            grayscale=GrayScaleAreaModelFactory.create(),
+        ),
+        {},
+        True,
+    ),
+    (
+        FixturePlateFactory.create(),
+        {},
+        True,
+    ),
+    (
+        FixtureFactory.create(
+            grayscale=GrayScaleAreaModelFactory.create(),
+            plates=(FixturePlateFactory.create(),),
+        ),
+        {},
+        True,
+    ),
+    (
+        FixtureFactory.create(
+            grayscale=GrayScaleAreaModelFactory.create(),
+        ),
+        {
+            "plates": (GridModelFactory.create(),),
+        },
+        False,
+    ),
+    (
+        FixtureFactory.create(
+            grayscale=GrayScaleAreaModelFactory.create(),
+        ),
+        {
+            "plates": (FixturePlateFactory.create(), None),
+        },
+        True,
+    ),
+))
+def test_validate(model: Model, updates: dict[str, Any], expect: bool):
```

Comment:
Maybe a few of these cases could do with a comment?

Reply:
(Here and elsewhere, but I'm not sure if it would help much.)

Reply:
added here at least